GermanGrib/rs-react
# React test context

1. Task: [link](https://github.com/rolling-scopes-school/tasks/blob/master/react/modules/module03/README.md)
2. Screenshot: <img width="1440" alt="Screenshot 2023-11-11 at 11 35 37" src="https://github.com/GermanGrib/rs-react/assets/112155209/13f4fca9-b074-483c-a7cd-4e603f20668a">
3. Done 11.11.2023 / deadline 13.11.2023
4. Score: 100 / 100


 - Student can get 100
  - [x] Custom state management using the Context API is implemented - 5 points
  - [x] The Search component value is stored in the context - 5 points
  - [x] The list of items received from the API is stored in the context - 5 points
  - [x] The context is used in components that need access to the data - 5 points
  - [x] React Testing Library and Jest or Vitest are added and configured - 10 points
  - [x] Test cases - 60 points (5 points per each)
1. Verify that the component renders the specified number of cards;
2. Check that an appropriate message is displayed if no cards are present.
3. Ensure that the card component renders the relevant card data;
4. Validate that clicking on a card opens a detailed card component;
5. Check that clicking triggers an additional API call to fetch detailed information.
6. Check that a loading indicator is displayed while fetching data;
7. Make sure the detailed card component correctly displays the detailed card data;
8. Ensure that clicking the close button hides the component.
9. Make sure the component updates URL query parameter when page changes.
10. Verify that clicking the Search button saves the entered value to the local storage;
11. Check that the component retrieves the value from the local storage upon mounting.
12. Ensure that the 404 page is displayed when navigating to an invalid route.
  - [x] Husky runs tests on pre-push - 10 points
